sdk/tests/conformance2/transform_feedback/simultaneous_access.html

```diff
@@ -0,0 +1,201 @@
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<canvas id="canvas" style="width: 50px; height: 50px;"> </canvas>
+<div id="console"></div>
+<script id="vshader" type="x-shader/x-vertex">#version 300 es
+in float in_value;
+out float out_value;
+
+void main() {
+   out_value = in_value * 2.;
+}
+</script>
+<script id="fshader" type="x-shader/x-fragment">#version 300 es
+precision mediump float;
+out vec4 dummy;
+uniform UniformBlock {
+  float fragment_value;
+};
+void main() {
+  dummy = vec4(fragment_value);
+}
+</script>
+<script>
+"use strict";
+description("This test verifies that simultaneous access to a buffer by transform feedback and other mechanisms is forbidden.");
+
+debug("");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+var gl = wtu.create3DContext(canvas, null, 2);
+
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+
+    runDrawArraysTest();
+    finishTest();
+}
+
+function runDrawArraysTest() {
+  const prog = wtu.setupTransformFeedbackProgram(gl, ["vshader", "fshader"],
+      ["out_value"], gl.SEPARATE_ATTRIBS,
+      ["in_value"]);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "linking transform feedback shader should not set an error");
+
+  const inLoc = 0;
+  const outLoc = 0;
+
+  const srcBuffer1 = createBuffer(gl, new Float32Array([1, 2, 3, 4]));
+  const srcVAO1 = createVAO(gl, srcBuffer1, inLoc);
+
+  const dstBuffer = createBuffer(gl, Float32Array.BYTES_PER_ELEMENT * 4);
+  const dstBuffer2 = createBuffer(gl, Float32Array.BYTES_PER_ELEMENT * 4);
+  const srcVAO2 = createVAO(gl, dstBuffer, inLoc);
+
+  const uniformBuffer = createBuffer(gl, new Float32Array([1, 0, 0, 0]));
+  const ubi = gl.getUniformBlockIndex(prog, "UniformBlock");
+  gl.uniformBlockBinding(prog, ubi, 0);
+  gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, uniformBuffer);
+
+  const tf = gl.createTransformFeedback();
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+  gl.useProgram(prog);
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, outLoc, dstBuffer);
+  // this binds the default (id = 0) TRANSFORM_FEEBACK buffer
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+  runFeedback(gl, prog, srcVAO1, tf);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "draw should be successful");
+
+  const expected = [2, 4, 6, 8];
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected);
+
+  // this should fail because the transform feedback's buffer #0 and the
+  // srcVAO2's buffer #0 are the same buffer
+  runFeedback(gl, prog, srcVAO2, tf);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as vertex attrib and tf simultaneously");
+
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected, "should be the same as before as nothing has executed");
+
+  gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, dstBuffer);
```

Comment:
It's still bound as `ARRAY_BUFFER` at this point, right? Shouldn't it be unbound from `ARRAY_BUFFER` so that doesn't interfere with the test of `UNIFORM_BUFFER`? Same in a few places below.

Reply:
Not sure how fragile this test is (for false positives), but to make it less fragile you could possibly make new buffers every time.